updating/updating-disconnected-cluster.adoc

```diff
@@ -1,23 +1,134 @@
 [id="updating-disconnected-cluster"]
-= Updating a disconnected cluster to a minor version
+= Updating a disconnected cluster
 include::modules/common-attributes.adoc[]
 :context: updating-disconnected-cluster
 
 toc::[]
 
-You can update, or upgrade, an {product-title} cluster that you installed in a
-restricted network to a minor version by using the web console.
+You can update, or upgrade a disconnected or fully air gapped {product-title} cluster using `oc` commandline.
 
 .Prerequisites
 
+* Have access to a conatiner registry in the disconnected environment to push and pull images. The container registry should be compatible with Docker registry API v2.
 * Have access to the cluster as a user with `admin` privileges.
 See xref:../authentication/using-rbac.adoc[Using RBAC to define and apply permissions].
 * Have a recent xref:../backup_and_restore/backing-up-etcd.adoc#backup-etcd[etcd backup] in case your upgrade fails and you must xref:../backup_and_restore/disaster_recovery/scenario-2-restoring-cluster-state.adoc#dr-restoring-cluster-state[restore your cluster to a previous state].
 
-//mirror module
+[IMPORTANT]
+====
+If you are upgrading from earlier release than 4.5.0, you must must use the manual method of creating configmap.
 
-include::modules/update-service-overview.adoc[leveloffset=+1]
+This is because `oc` command enhancement is not available in previous releases.
+====
 
-include::modules/understanding-upgrade-channels.adoc[leveloffset=+1]
+[id="update-airgapped-env"]
+== Updating an air gapped {product-title} cluster
+
+An air gapped environment is the one in which your cluster nodes cannot access the internet. For this reason you must mirror the images to a filesystem disconnected from that environment and then bring that host or removable media across that gap. It may also be the case that there are multiple clusters within the air gapped network. In such a case it makes sense to configure a central registry in the air gapped environment from which every cluster can pull required images for upgrade.
+
+.Procedure
+
+. Findout if there is link:https://access.redhat.com/solutions/4583231[an update path] exists between your current version and the version you want the cluster to move.
+
+. Mirror the version images to the internal container registry.
+
+** Connect the removable media to a system connected to the internet.
+** Mirror the images and configuration manifests to a directory on the removable media.
++
+----
+$ oc adm release mirror "--to-dir=${REMOVABLE_MEDIA_PATH}/mirror" quay.io/openshift-release-dev/ocp-release:4.3.0-x86_64
+----
++
+
+** Take the media to the disconnected environment and upload the images to the local container registry
++
+----
+$ oc image mirror "--from-dir=${REMOVABLE_MEDIA_PATH}/mirror" file://openshift/release:4.3.0* REGISTRY/REPOSITORY
```

Comment:
`REGISTRY` and `REPOSITORY` should be actual environment variables and be declared with docs around their semantics (like [this][1], although ideally without the `export`, #22008).

[1]: https://github.com/openshift/openshift-docs/blame/4abd206b42d38e5bd66423d780c4c900b395e8ff/modules/installation-mirror-repository.adoc#L35-L54

Reply:
I am not a huge fan of the shell scripting approach, but I understand why people do it.  The goal is NOT to have fully executable scripts in our docs.  The goal is not to need that.  Having to bash is a gap in the product experience - we shouldn't just paper over those.

Reply:
>  Having to bash is a gap in the product experience - we shouldn't just paper over those.

I don't think this is a product gap; we're just using shell as a convenient, familiar placeholder syntax here, and uesrs can plug in their chosen registry and repository without bothering with actual shell variables if they want.  The alternative of using placeholders like `REGISTRY` (instead of `$REGISTRY`) is just defining our own, non-standard placeholder syntax, and I see no benefit to doing that.

Reply:
I agree with @smarterclayton  about having a bunch of shell scripts is the ideal user experience we want for our customers. I am on board with figuring out how we can get rid of the shell scripts and may be provide a better user experience in future.

Reply:
@LalatenduMohanty  Not specifically related to this PR, but seems like maybe we should be building a workflow around for whatever graph viewer we come up with in OCM. I.e. you tell us what version you're on; we recommend some options; you choose a version; and then we help you create whatever needs creating so that you can go off and do the mirroring/upgrade. cc @katherinedube

Reply:
I would [still][1] rather punt on target discovery, and have this PR scoped to the mechanics of post-discovery application.

[1]: https://github.com/openshift/openshift-docs/pull/21993#discussion_r423287574